File: 53014670源码/backend/app/schemas/recommend.py
from pydantic import BaseModel, Field


class PoiItem(BaseModel):
    id: str
    name: str
    lat: float
    lng: float
    description: str
    tags: list[str]
    visitMinutes: int = 30


class RouteItem(BaseModel):
    routeId: str
    name: str
    description: str
    pois: list[PoiItem]
    estimatedDuration: int
    highlights: list[str]
    matchScore: float = 0.0


class RecommendRoutesData(BaseModel):
    routes: list[RouteItem]
    preferences: list[str] = Field(default_factory=list)
    scenicAreaId: str = "lingshan_scenic"
    scenicAreaName: str = "灵山胜境"


class RecommendRoutesRequest(BaseModel):
    preference: list[str] = Field(
        default_factory=list,
        description='兴趣标签，如 ["history", "nature"]',
        examples=[["history", "nature"]],
    )

File: 53014670源码/web-client/src/types/recommend.ts
export interface PoiItem {
  id: string
  name: string
  lat: number
  lng: number
  description: string
  tags: string[]
  visitMinutes: number
}

export interface RouteItem {
  routeId: string
  name: string
  description: string
  pois: PoiItem[]
  estimatedDuration: number
  highlights: string[]
  matchScore: number
}

export interface RecommendRoutesData {
  routes: RouteItem[]
  preferences: string[]
}

export interface RecommendRoutesRequest {
  preference: string[]
}

export interface PreferenceOption {
  id: string
  label: string
  icon: string
}

export const PREFERENCE_OPTIONS: PreferenceOption[] = [
  { id: 'history', label: '历史', icon: '🏛️' },
  { id: 'culture', label: '文化', icon: '📜' },
  { id: 'nature', label: '自然', icon: '🌿' },
  { id: 'photo', label: '摄影', icon: '📷' },
  { id: 'family', label: '亲子', icon: '👨‍👩‍👧' },
  { id: 'adventure', label: '探险', icon: '🥾' },
]

export const TAG_LABELS: Record<string, string> = {
  history: '历史',
  culture: '文化',
  nature: '自然',
  photo: '摄影',
  family: '亲子',
  adventure: '探险',
}
